lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -3013,6 +3028,90 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
             . Map.toList
             . foldr (uncurry (Map.insertWith (<>))) Map.empty
 
+encryptionNonce :: ByteString
+encryptionNonce = "encrypt-data"
+
+-- When encryption is enabled we do the following:
+-- (a) toJSON on metadata,
+-- (b) encrypt it
+-- (c) and create new tx metadata
+--     (0, TxMetaBytes chunk1)
+--     (1, TxMetaBytes chunk2)
+--     ....
+--     (N, TxMetaBytes chunkN)
+--     until encrypted metadata is accommodated
+toMetadataEncrypted
+    :: ApiEncryptMetadata
+    -> TxMetadataWithSchema
+    -> Cardano.TxMetadata
```

Comment:
Metadata encryption is not a HTTP API concern, and therefore shouldn't be placed in the `lib/wallet/api/http/**`.